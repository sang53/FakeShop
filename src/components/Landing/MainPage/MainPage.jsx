import Carousel from "../Carousel/Carousel.jsx";

export default function MainPage() {
  return (
    <>
      <div className="flex-column-center gap-1rem text-center">
        <h1>Welcome to Fake Shopping Project!</h1>
        <p>Feel free to browse by category or product!</p>
        <p>
          This project was made with React & React-router as part of The Odin
          Project
        </p>
        <p>Here are some random products to view below!</p>
      </div>
      <Carousel />
    </>
  );
}
