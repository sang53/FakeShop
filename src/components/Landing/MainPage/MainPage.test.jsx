import { render } from "@testing-library/react";
import { expect, test, vi } from "vitest";
import MainPage from "./MainPage";

vi.mock("../Carousel/Carousel", () => ({
  default: () => <div data-testid="mock-carousel">Carousel</div>,
}));

test("Main Landing Page elements display as expected", () => {
  const { container } = render(<MainPage />);
  expect(container).toMatchSnapshot();
});
