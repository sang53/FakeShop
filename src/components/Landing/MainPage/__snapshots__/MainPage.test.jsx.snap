// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Main Landing Page elements display as expected 1`] = `
<div>
  <div
    class="flex-column-center gap-1rem text-center"
  >
    <h1>
      Welcome to Fake Shopping Project!
    </h1>
    <p>
      Feel free to browse by category or product!
    </p>
    <p>
      This project was made with React & React-router as part of The Odin Project
    </p>
    <p>
      Here are some random products to view below!
    </p>
  </div>
  <div
    data-testid="mock-carousel"
  >
    Carousel
  </div>
</div>
`;
